test/cmpd_dset.c

```diff
@@ -108,6 +108,835 @@ typedef struct {
 #define NY          2000U
 #define PACK_NMEMBS 100
 
+static void  initialize_stype1(unsigned char *buf, size_t num);
+static void  initialize_stype2(unsigned char *buf, size_t num);
+static void  initialize_stype3(unsigned char *buf, size_t num);
+static void  initialize_stype4(unsigned char *buf, size_t num);
+static hid_t create_stype1(void);
+static hid_t create_stype2(void);
+static hid_t create_stype3(void);
+static hid_t create_stype4(void);
+static int   compare_data(void *src_data, void *dst_data, hbool_t src_subset);
+static int   compare_stype4_data(void *expect_buf, void *rbuf);
+
+/*-------------------------------------------------------------------------
+ * Function:    compare_stype4_data
+ *
+ * Purpose:     Compare data (stype4) read in rbuf with expected data
+ *              in expect_buf.
+ *
+ * Return:      Success:        0
+ *
+ *              Failure:        negative
+ *
+ *-------------------------------------------------------------------------
+ */
+static int
+compare_stype4_data(void *expect_buf, void *rbuf)
+{
+    int      i;
+    unsigned err = FALSE;
+
+    for (i = 0; i < (int)(NX * NY); i++) {
+        stype4 *s1_ptr;
+        stype4 *s2_ptr;
+        s1_ptr = ((stype4 *)expect_buf) + i;
+        s2_ptr = ((stype4 *)rbuf) + i;
+
+        if (s1_ptr->a != s2_ptr->a) {
+            printf("a is different\n");
```

Comment:
Were these debugging printfs, or do you still want these print statements?

Reply:
Yes, I will do some cleanups of the tests today.